packages/typespec-ts/test/modularUnit/anonymousModel.spec.ts

```diff
@@ -0,0 +1,671 @@
+            throw result.body;
+          }
+          return result.body;
+        }
+        export async function read(
+          context: Client,
+          options: ReadOptions = { requestOptions: {} }
+        ): Promise<${returnType}> {
+          const result = await _readSend(
+            context,
+            options
+          );
+          return _readDeserialize(result);
+        }`,
+          true
+        );
+      }
+      it("should map empty anonymous model({}) => {}", async () => {
+        const tspContent = `
+        op read(): {};
+        `;
+        // No models.ts file generated
+        assert.isUndefined(await emitModularModelsFromTypeSpec(tspContent));
+        const operationFiles = await emitModularOperationsFromTypeSpec(
+          tspContent
+        );
+        assert.equal(operationFiles?.length, 1);
+        // Generate the operations.ts file with empty model
+        verifyReturnTypeAsEmpty(operationFiles?.[0]?.getFullText()!, "{}");
+      });
+
+      it("should map empty named model(A {}) => A {}", async () => {
+        const tspContent = `
+        model PublishResult {
+        }
+        op read(): PublishResult;
+        `;
+        // Empty models generated in models.ts
+        const modelFile = await emitModularModelsFromTypeSpec(tspContent);
+        assertEqualContent(
+          modelFile?.getFullText()!,
+          `
+        export interface PublishResult {}
+        `
+        );
+        const operationFiles = await emitModularOperationsFromTypeSpec(
+          tspContent
+        );
+        assert.ok(operationFiles);
+        assert.equal(operationFiles?.length, 1);
+        // Model name referred in operations.ts
+        verifyReturnTypeAsEmpty(
+          operationFiles?.[0]?.getFullText()!,
+          "PublishResult"
+        );
+      });
+
+      it("should return non-empty anonymous  model({ ... })", async () => {});
+    });
+    describe("happens as a property in response body", async () => {
+      it("should map empty anonymous  model({}) => {} & empty named model(A {}) => A {}", async () => {
+        const tspContent = `
+        model EmptyModel {
+        }
+        model ReturnBody {
+          emptyAnomyous: {};
+          emptyAnomyousArray: {}[];
+          emptyAnomyousDict: Record<{}>;
+          emptyModel: EmptyModel;
+          emptyModelArray: EmptyModel[];
+          emptyModelDict: Record<EmptyModel>;
+        }
+        op read(): ReturnBody;
+        `;
+        const modelFile = await emitModularModelsFromTypeSpec(tspContent);
+        console.log(modelFile?.getFullText());
+        assertEqualContent(
+          modelFile?.getFullText()!,
+          `
+        export interface ReturnBody {
+          emptyAnomyous: {};
+          emptyAnomyousArray: {}[];
+          emptyAnomyousDict: Record<string, {}>;
+          emptyModel: EmptyModel;
+          emptyModelArray: EmptyModel[];
+          emptyModelDict: Record<string, EmptyModel>;
```

Comment:
This is the generated model for empty anonymous/named model now we don't treat them as `Record<string, any>`.